manufacturing-client/src/main.rs

```diff
@@ -295,11 +298,70 @@ async fn get_mfg_info(mfg_string_type: MfgStringType) -> Result<CborSimpleType>
         return Ok(CborSimpleType::Text(mfg_info.into_string().unwrap()));
     }
     match mfg_string_type {
-        MfgStringType::SerialNumber => {
-            let serial = fs::read_to_string("/sys/devices/virtual/dmi/id/product_serial")
+        MfgStringType::SerialNumberExt => {
+            let serial_dmi = fs::read_to_string("/sys/devices/virtual/dmi/id/product_serial")
                 .or_else(|_| fs::read_to_string("/sys/devices/virtual/dmi/id/chassis_serial"))
                 .context("Error determining system serial number")?;
-            Ok(CborSimpleType::Text(serial))
+            let mut arg: String;
+            let output_ifaces = Command::new("ls")
+                .arg("/sys/class/net/")
+                .stdout(std::process::Stdio::piped())
+                .stderr(std::process::Stdio::piped())
+                .output()
+                .expect("Failed to execute command");
+
+            let val = String::from_utf8_lossy(&output_ifaces.stdout).to_string();
+            let mut serial_mac: String = "".to_string();
+            for interface in val.split('\n') {
+                arg = "/sys/class/net/".to_owned() + interface + "/address";
+                let output = Command::new("/bin/cat")
+                    .args([":", &arg])
+                    .stdout(std::process::Stdio::piped())
+                    .stderr(std::process::Stdio::piped())
+                    .output()
+                    .expect("Failed to execute grep command");
+                let mac = String::from_utf8_lossy(&output.stdout)
+                    .trim_end()
+                    .to_string();
+                let mac_regex = Regex::new(r"^([0-9A-Fa-f]{2}[:-]){5}([0-9A-Fa-f]{2})$").unwrap();
+                if mac_regex.is_match(&mac) {
+                    let mut non_zero_mac = false;
+                    for byte in mac.split(':') {
+                        let byte = u8::from_str_radix(byte, 16).unwrap();
+                        if byte != 0 {
+                            non_zero_mac = true;
+                            break;
+                        }
+                    }
+                    if non_zero_mac {
+                        serial_mac = serial_mac
```

Comment:
Maybe a bit more ergonomic:

```
                        write!(serial_mac, "IFACE_{interface}={mac};").unwrap();
```

I'd unwrap here because this can't actually fail (except on OOM) but Rust panic on that anyways.